apis/v1beta1/vspheremachine_types.go

```diff
@@ -42,6 +43,34 @@ type VSphereMachineSpec struct {
 	// FailureDomain is the failure domain unique identifier this Machine should be attached to, as defined in Cluster API.
 	// For this infrastructure provider, the name is equivalent to the name of the VSphereDeploymentZone.
 	FailureDomain *string `json:"failureDomain,omitempty"`
+
+	// PowerOffMode describes the desired behavior when powering off a VM.
+	//
+	// There are three, supported power off modes: hard, soft, and
+	// trySoft. The first mode, hard, is the equivalent of a physical
+	// system's power cord being ripped from the wall. The soft mode
+	// requires the VM's guest to have VM Tools installed and attempts to
+	// gracefully shut down the VM. Its variant, trySoft, first attempts
+	// a graceful shutdown, and if that fails or the VM is not in a powered off
+	// state after reaching the GuestSoftPowerOffTimeout, the VM is halted.
+	//
+	// If omitted, the mode defaults to trySoft.
+	//
+	// +optional
+	// +kubebuilder:default=trySoft
```

Comment:
Note: this changes the behaviour of CAPV with the next version.

So this should get highlighted in a release note. Please make sure the PR description covers that.